feathr_project/feathr/spark_provider/_localspark_submission.py

```diff
@@ -0,0 +1,224 @@
+                cfg["spark.jars.packages"] = ",".join([default_packages, FEATHR_MAVEN_ARTIFACT])
+
+            if not python_files:
+                # This is a JAR job
+                # Azure Synapse/Livy doesn't allow JAR job starts from Maven directly, we must have a jar file uploaded.
+                # so we have to use a dummy jar as the main file.
+                logger.info(f"Main JAR file is not set, using default package '{FEATHR_MAVEN_ARTIFACT}' from Maven")
+                # Use the no-op jar as the main file
+                # This is a dummy jar which contains only one `org.example.Noop` class with one empty `main` function which does nothing
+                current_dir = Path(__file__).parent.resolve()
+                main_jar_path = os.path.join(current_dir, "noop-1.0.jar")
+                args.append("--packages %s" % cfg["spark.jars.packages"])
+                args.append('--class %s' % main_class_name)
+                args.append('%s '%main_jar_path)
+            else:
+                args.append("--packages %s" % cfg["spark.jars.packages"])
+                # This is a PySpark job, no more things to 
+                if python_files.__len__() > 1:
+                    args.append('--py-files %s' % ",".join(python_files[1:]))
+                print(python_files)
+                args.append('%s '%python_files[0])
+        else:
+            args.append('--class %s' % main_class_name)
+            args.append('%s '%main_jar_path)
+
+        cmd = ' '.join(args) + ' '.join(arguments)
+
+        log_a = open(f'{self.log_path}_{self.spark_job_num}.txt' , 'a')     
+        proc = Popen(split(cmd), shell=False, stdout=log_a, stderr=STDOUT)
```

Comment:
why did you join and then split `cmd`?

Reply:
Because based on my tests, the argument list cannot be directly used by Popen. The split function will be a more reliable choice.